wallets/react/src/provider.tsx

```diff
@@ -1,228 +1,15 @@
-        }
-      });
-
-      return providers;
-    },
-    getWalletInfo(type) {
-      const wallet = wallets.get(type);
-      if (!wallet) {
-        throw new Error(`You should add ${type} to provider first.`);
-      }
-
-      /*
-       * Get wallet info could be used in render methods to show wallets data
-       * So, addWalletRef method shouldn't be called in this method
-       */
-
-      return wallet.actions.getWalletInfo(props.allBlockChains || []);
-    },
-    getSigners(type) {
-      const wallet = wallets.get(type);
-
-      if (!wallet) {
-        throw new Error(`You should add ${type} to provider first.`);
-      }
-      const walletInstance = getWalletInstance(wallet);
-      const provider = walletInstance.provider;
-      const result = walletInstance.getSigners(provider);
-
-      return result;
-    },
-  };
-
-  useEffect(() => {
-    wallets.forEach((wallet) => {
-      const walletInstance = getWalletInstance(wallet);
-      const runOnInit = () => {
-        if (walletInstance.onInit) {
-          walletInstance.onInit();
-        }
-      };
-
-      const initWhenPageIsReady = (event: Event) => {
-        if (
-          event.target &&
-          (event.target as Document).readyState === 'complete'
-        ) {
-          runOnInit();
-
-          document.removeEventListener('readystatechange', initWhenPageIsReady);
-        }
-      };
-
-      // Try to run, maybe it's ready.
-      runOnInit();
-
-      /*
-       * Try again when the page has been completely loaded.
-       * Some of wallets, take some time to be fully injected and loaded.
-       */
-      document.addEventListener('readystatechange', initWhenPageIsReady);
-    });
-  }, []);
-
-  useEffect(() => {
-    const allBlockChains = props.allBlockChains;
-    if (allBlockChains) {
-      wallets.forEach((wallet) => {
-        const walletInstance = getWalletInstance(wallet);
-        const walletInfo = walletInstance.getWalletInfo(
-          props.allBlockChains || []
-        );
-        walletInstance.setInfo({
-          supportedBlockchains: walletInfo.supportedChains,
-          isContractWallet: !!walletInfo.isContractWallet,
-        });
-      });
-    }
-  }, [props.allBlockChains]);
-
-  useEffect(() => {
-    wallets.forEach((wallet) => {
-      const walletInstance = getWalletInstance(wallet);
-      walletInstance.setHandler(
-        makeEventHandler(dispatch, props.onUpdateState)
-      );
-    });
-  }, [props.onUpdateState]);
-
-  useEffect(() => {
-    const shouldTryAutoConnect =
-      props.allBlockChains &&
-      props.allBlockChains.length &&
-      props.autoConnect &&
-      !autoConnectInitiated.current;
-
-    if (shouldTryAutoConnect) {
-      autoConnectInitiated.current = true;
-      void (async () => {
-        await autoConnect(wallets, getWalletInstance);
-      })();
-    }
-  }, [props.autoConnect, props.allBlockChains]);
+  const legacyApi = useLegacyProviders(props);
```

Comment:
I moved all the logics into `useLegacyProviders`. In a separate PR, I will add hub providers as a hook as well and there will be a adapter to make both of them work without breaking anything.